File: ClientApp/ClientApp/Models/Responses/GetRoomsResponse.cs
using Newtonsoft.Json;
using ClientApp.Models.ResponseErrors;
namespace ClientApp.Models.Responses
{
    class GetRoomsResponse : Response<NoResponseErrors>
    {
        private GetRoomsResponse() { }

        [JsonProperty]
        public List<RoomPreview> Rooms { get; private set; }
    }
}

File: M_Trivia/M_Trivia/GetRoomsResponse.cpp
#include "GetRoomsResponse.hpp"

#include "json.hpp"



GetRoomsResponse::GetRoomsResponse(unsigned int status, const std::vector<RoomPreview>& rooms) : 
	Response(status), rooms(rooms) 
{

}


nlohmann::json GetRoomsResponse::getJson() const {
	nlohmann::json j = Response::getJson();
	j["Rooms"] = rooms; // Uses the to_json for PersonalStatistics
	return j;
}

